packages/grafana-ui/src/components/Combobox/useOptions.ts

```diff
@@ -56,6 +58,27 @@ export function useOptions<T extends string | number>(rawOptions: AsyncOptions<T
   // told it for async options loading anyway.
   const [userTypedSearch, setUserTypedSearch] = useState('');
 
+  const addCustomValue = useMemo(
```

Comment:
Can this be a `useCallback`?

Reply:
Why would this be `useCallback`? I used `useMemo` because I thought we would want to cache the result 🤔

Reply:
Because you are returning a function. The options are being cached in `finalOptions` anyways.